File: Short/bubbleSort.r
bubble_sort <- function(arr) {
  n <- length(arr)
  for (i in 1:n) {
    for (j in (i+1):n) {
      if (arr[i] > arr[j]) {
        temp <- arr[i]
        arr[i] <- arr[j]
        arr[j] <- temp
      }
    }
  }
  return(arr)
}

# Taking user input
n <- as.integer(readline("How many numbers do you want? "))
arr <- numeric(n)

cat("Enter numbers:\n")
for (i in 1:n) {
  arr[i] <- as.integer(readline())
}

sorted_arr <- bubble_sort(arr)
cat("Sorted Array:", sorted_arr, "\n")

File: Short/BubbleSort.java
// package Short;
import java.util.*;

public class BubbleSort {
    public static void bubbleSort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (arr[i] > arr[j]) {
                    int temp = arr[i];
                    arr[i] = arr[j];
                    arr[j] = temp;
                }
            }
        }
        System.out.println("Sorted Array is: ");
        for (int num : arr) {
            System.out.print(num + " ");
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("How many numbers do you want? ");
        int n = sc.nextInt();
        
        int[] arr = new int[n];
        System.out.println("Enter numbers:");
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        
        bubbleSort(arr);
        sc.close();
    }
}
